src/lib/y2network/sysconfig/interfaces_writer.rb

```diff
@@ -77,15 +80,41 @@ def update_udevd(interfaces)
         reload_udev_rules
       end
 
+      # Updates the given iface udev rule depending on the renaming mechanism
+      # selected
+      #
+      # @param iface [Interface] Interface to update the udev rule for
+      # @return [UdevRule] udev rule
+      def update_renaming_udev_rule(iface)
+        case iface.renaming_mechanism
+        when :mac
+          iface.udev_rule.rename_by_mac(iface.name, iface.hardware.mac)
+        when :bus_id
+          iface.udev_rule.rename_by_bus_id(iface.name, iface.hardware.busid,
+            iface.hardware.dev_port)
+        end
+
+        iface.udev_rule
+      end
+
+      # Writes down the current interfaces udev rules and the custom rules that
+      # were present when read and that are still valid
+      #
+      # @see Y2Network::UdevRule#write_net_rules
       def update_renaming_udev_rules(interfaces)
-        udev_rules = interfaces.map { |i| renaming_udev_rule_for(i) }.compact
+        udev_rules = interfaces.map do |iface|
+          update_renaming_udev_rule(iface) if iface.udev_rule
+          iface.udev_rule ||= renaming_udev_rule_for(iface)
```

Comment:
this is a bit strange. If there is already udev_rule it will be updated. And if there is not yet such udev_rule it is created. Why not having it in one method?

Reply:
also I a bit expect to be a method in interface and not only in writer.

Reply:
Originally I put the logic in one method, but was not sure about merging the logic for new rules and for existing ones.